src/pets/pets.service.ts

```diff
@@ -0,0 +1,150 @@
+import { Model } from 'mongoose';
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Cat } from './interfaces/cat.interface';
+import { Dog } from './interfaces/dog.interface';
+import { Owner } from './interfaces/owner.interface';
+import { getTotalWeight } from './weight.helper';
+import { CreateCatDto } from './dto/create.cat.dto';
+import { CreateDogDto } from './dto/create.dog.dto';
+
+@Injectable()
+export class PetsService {
+  constructor(
+    @InjectModel('Cat') private readonly catModel: Model<Cat>,
+    @InjectModel('Dog') private readonly dogModel: Model<Dog>,
+    @InjectModel('Owner') private readonly ownerModel: Model<Owner>,
+  ) {}
+
+  async addCat(createCatDto: CreateCatDto): Promise<Cat> {
+    var createdCat = new this.catModel(createCatDto);
```

Comment:
please, kindly use ES6.
```suggestion
    const createdCat = new this.catModel(createCatDto);
```